### PR TITLE
Update substrate master

### PR DESCRIPTION
**Pull request by thiolliere**
Syntax of deposit_event have changed.

Ready to merge

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ use sr_staking_primitives::SessionIndex;
 use srml_support::{
 	parameter_types, construct_runtime, traits::{SplitTwoWays, Currency}
 };
-use im_online::AuthorityId as ImOnlineId;
+use im_online::sr25519::{AuthorityId as ImOnlineId};
```
Comment (thiolliere) [inline runtime/src/lib.rs]: we use this one in node runtime in substrate though I don't know which one we want here

**Comment by thiolliere** [comment]
test failure:
```

---- adder::execute_good_chain_on_parent stdout ----
thread 'adder::execute_good_chain_on_parent' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout', src/libcore/result.rs:999:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

---- adder::execute_good_on_parent stdout ----
thread 'adder::execute_good_on_parent' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout', src/libcore/result.rs:999:5

---- adder::processes_messages stdout ----
thread 'adder::processes_messages' panicked at 'called `Result::unwrap()` on an `Err` value: Timeout', src/libcore/result.rs:999:5


failures:
    adder::execute_good_chain_on_parent
    adder::execute_good_on_parent
    adder::processes_messages

test result: FAILED. 4 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out
```

File: runtime/src/lib.rs (hunk 1 above)
Comment (kianenigma) [reply runtime/src/lib.rs]: might help 

https://github.com/paritytech/substrate/pull/3500#discussion_r318454237

**Comment by gavofyork** [comment]
@thiolliere looks like it builds ok.

**Comment by thiolliere** [comment]
> @thiolliere looks like it builds ok. Once this is ready please backport changes to `v0.5` branch.

I'm not sure I understand, this PR is ready to me, I'm working on the backport now

**Pull request merged**